app/v2/controllers/V2MovementsController.scala

```diff
@@ -352,8 +352,19 @@ class V2MovementsControllerImpl @Inject() (
               downloadUrl   <- handleUpscanSuccessResponse(upscanResponse)
               objectSummary <- objectStoreService.addMessage(downloadUrl, movementId, messageId).asPresentation
               messageUpdate = MessageUpdate(MessageStatus.Processing, Some(ObjectStoreURI(objectSummary.location.asUri)))
-              update <- persistenceService.updateMessage(movementId, messageId, messageUpdate).asPresentation
-            } yield update).fold[Result](
+
+              _ <- persistenceService.updateMessage(movementId, messageId, messageUpdate).asPresentation
+
+              sendMessage <- routerService
+                .sendLargeMessage(
+                  MessageType.DeclarationData, // Only IE015 is supported as of now for large message
```

Comment:
we should be extracting the message type here, if it's not a MessageType.DeclarationData then we can fail the request - I think it would be wrong to hard-code it here since it is technically possible to not send that specific type

Reply:
@drnaylor-oc / @liandra-giacomine  Not sure if I misunderstood, but in morning call Dan said that we can hardcode ``` MessageType.DeclarationData ``` as if you can see for createDepartureDeclaration we are not validating MessageType. just simple passing that  ``` MessageType.DeclarationData  ```

if we want to validate MessageType here then I think we need to make get call to object store get the source and then extract the messageType using existing logic  ``` xmlParsingService.extractMessageType(source, messageTypeList).asPresentation ```  -- messageTypeList = Seq(MessageType.DeclarationData)

is that what we expecting here?

Reply:
Oh fair enough, we can do that in a later ticket then